src/main/java/online/shop/dao/jdbc/ProductDaoJdbsImpl.java

```diff
@@ -0,0 +1,111 @@
+package online.shop.dao.jdbc;
+
+import java.sql.ResultSet;
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.sql.PreparedStatement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import online.shop.dao.ProductDao;
+import online.shop.exception.DataProcessingException;
+import online.shop.lib.Dao;
+import online.shop.model.Product;
+import online.shop.util.ConnectionUtil;
+
+@Dao
+public class ProductDaoJdbsImpl implements ProductDao {
+    @Override
+    public Product create(Product item) {
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            String query = "INSERT INTO product_table (name, price) VALUES (?, ?);";
+            PreparedStatement statement =
+                    connection.prepareStatement(query, Statement.RETURN_GENERATED_KEYS);
+            statement.setString(1, item.getName());
+            statement.setDouble(2, item.getPrice());
+            statement.executeUpdate();
+            ResultSet resultSet = statement.getGeneratedKeys();
+            if (resultSet.next()) {
+                item.setId(resultSet.getLong("product_id"));
+            }
+            return item;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't add item!", e);
+        }
+    }
+
+    @Override
+    public Optional<Product> getById(Long id) {
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            String query = "SELECT * FROM product_table WHERE product_id = ? AND deleted=false;";
+            PreparedStatement statement =
+                    connection.prepareStatement(query);
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                return Optional.of(getProductFromResultSet(resultSet));
+            }
+            return Optional.empty();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't find product by id!", e);
```

Comment:
```suggestion
            throw new DataProcessingException("Can't find product by id!" + id, e);
```
Maybe specifying the id that caused a problem in the exception message could help you debug later)